data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -61,15 +72,35 @@ public KeyValueProcessor(final PluginMetrics pluginMetrics, final KeyValueProces
                 throw new PatternSyntaxException("field_delimiter_regex is not a valid regex string", keyValueProcessorConfig.getFieldDelimiterRegex(), -1);
             }
 
+            if (keyValueProcessorConfig.getRecursive()) {
+                if (Pattern.compile(keyValueProcessorConfig.getFieldDelimiterRegex()).matcher(delimiterBracketCheck).matches()) {
+                    throw new IllegalArgumentException("The set field delimiter regex cannot contain brackets while you are trying to recurse.");
+                }
+                if (keyValueProcessorConfig.getFieldDelimiterRegex().length() != 1) {
+                    throw new IllegalArgumentException("The set field delimiter is limited to one character only.");
+                }
+            }
+
             fieldDelimiterPattern = Pattern.compile(keyValueProcessorConfig.getFieldDelimiterRegex());
         } else {
             String regex;
             if (keyValueProcessorConfig.getFieldSplitCharacters().isEmpty()) {
                 regex = KeyValueProcessorConfig.DEFAULT_FIELD_SPLIT_CHARACTERS;
             } else {
+                if (keyValueProcessorConfig.getRecursive()) {
+                    if (keyValueProcessorConfig.getFieldSplitCharacters().length() != 1) {
```

Comment:
Since we are restricting the field split character to be single only when recursive is true, you could combine the nested if condition into 1.